### PR TITLE
options view: volume functionality

### PR DESCRIPTION
**Pull request by arazulu**
- created volumes toggle view: rect svg w/ text w/ mouseover effects
- refer to JIRA ticket JOS-50

### CODE REVIEW

File: src/renderer/components/DependsOnView.tsx [hunk 1]
```diff
@@ -412,6 +412,96 @@ const DependsOnView: React.FC<Props> = ({
...
+              return vText.style('visibility', 'visible');
+            })
+            .on('mouseout', () => {
+              !onClick
+                ? vText.style('visibility', 'hidden')
+                : vText.style('visibility', 'visible');
+            })
+            .on('click', () => {
+              if (!onceClicked) {
+                onClick = true;
```
Comment (michaelbayday) [inline src/renderer/components/DependsOnView.tsx]: can this logic be shortened?

Comment (michaelbayday) [reply src/renderer/components/DependsOnView.tsx]: maybe via ternary operators or something

File: src/renderer/components/DependsOnView.tsx [hunk 2]
```diff
@@ -412,6 +412,96 @@ const DependsOnView: React.FC<Props> = ({
...
+            .attr('width', width)
+            .attr('height', height)
+            .attr('x', x)
+            .attr('y', y + i * 12)
+            .on('mouseover', () => {
+              return vText.style('visibility', 'visible');
+            })
+            .on('mouseout', () => {
+              !onClick
+                ? vText.style('visibility', 'hidden')
```
Comment (michaelbayday) [inline src/renderer/components/DependsOnView.tsx:464]: can this be handled with class and styled within sass file?

File: src/renderer/components/DependsOnView.tsx [hunk 3]
```diff
@@ -412,6 +412,96 @@ const DependsOnView: React.FC<Props> = ({
...
+            .attr('dy', y + (i + 1) * 11)
+            .style('visibility', 'hidden');
+          // store d3 object in volumes text array
+          volumeText.push(vText);
+        });
+      });
+    }
+
+    return () => {
+      // before unmounting, if volumes option was on, remove the volumes
```
Comment (michaelbayday) [inline src/renderer/components/DependsOnView.tsx:491]: good job with unmounting!

File: src/renderer/components/DependsOnView.tsx [hunk 4]
```diff
@@ -412,6 +412,96 @@ const DependsOnView: React.FC<Props> = ({
...
+                vString.slice(0, vString.indexOf(':')),
+              );
+              return slicedVString;
+            })
+            .attr('width', width)
+            .attr('height', height)
+            .attr('x', x)
+            .attr('y', y + i * 12)
+            .on('mouseover', () => {
+              return vText.style('visibility', 'visible');
```
Comment (michaelbayday) [inline src/renderer/components/DependsOnView.tsx:460]: can this be handled with class and styled within sass file?

File: src/renderer/components/DependsOnView.tsx [hunk 5]
```diff
@@ -412,6 +412,96 @@ const DependsOnView: React.FC<Props> = ({
...
+        .selectAll<SVGGElement, SNode>('g')
+        .filter((d: SNode) => d.volumes.length > 0);
+
+      // iterate through all nodes with volumes
+      nodesWithVolumes.each(function(d: SNode) {
+        const node = this;
+        // iterate through all volumes of node
+        d.volumes.forEach((vString, i) => {
+          let onClick = false;
+          let onceClicked = false;
```
Comment (michaelbayday) [inline src/renderer/components/DependsOnView.tsx:443]: wow! closure 8)

File: src/renderer/components/DependsOnView.tsx (hunk 1 above)
Comment (arazulu) [reply src/renderer/components/DependsOnView.tsx]: shortened with
 ```
.on('click', () => {
              onceClicked = !onceClicked;
              onClick = onceClicked;
            });
```

Comment (michaelbayday) [reply src/renderer/components/DependsOnView.tsx]: nice!!

File: src/renderer/components/DependsOnView.tsx (hunk 2 above)
Comment (arazulu) [reply src/renderer/components/DependsOnView.tsx:464]: not sure how to do this, if changing the class name styles wouldn't all the text node change and not that individual text node?

Comment (michaelbayday) [reply src/renderer/components/DependsOnView.tsx:464]: can you just append a string to the classname that would process as hidden? i'm not sure but i feel like it's possible

**Review by michaelbayday — APPROVED** [review]

**Pull request merged**